src/shim/binary_spinning_sem.cc

```diff
@@ -24,10 +17,11 @@ void BinarySpinningSem::post() {
 }
 
 void BinarySpinningSem::wait(bool spin) {
+    ssize_t spin_ctr = 0;
     // Based loosely on
     // https://probablydance.com/2019/12/30/measuring-mutexes-spinlocks-and-how-bad-the-linux-scheduler-really-is/.
     if (spin) {
-        while (_spin_ctr++ < _thresh) {
+        while (_thresh < 0 || spin_ctr++ < _thresh) {
```

Comment:
Probably not worth addressing in this PR, but I bet some use of `__builtin_expect` might give a good hint to the platform -- both these conditions are low entropy

Reply:
Hmm, yeah I'm a bit wary of such hints in general. Ironically it's not clear we have a reason to highly optimize the straight-line execution cost of this loop, since making the loop tighter just means we'll iterate more times.

Making it faster or slower does interact with `SHD_GATE_SPIN_MAX`, though. I've been wondering whether we'd be better off using RDTSC inside the loop and basing the max off of cycle count instead of loop count so that we can decouple "how long to spin" from "how many instructions ended up inside the loop in this compilation".

Either way, yeah let's save for another PR :)